vllm/model_executor/guided_logits_processors.py

```diff
@@ -0,0 +1,140 @@
+# Make vLLM compatible with Outlines' structured generation.
+#
+#  _______________________________
+# / Don't want to self-host?      \
+# \ Try .json at http://dottxt.co /
+#  -------------------------------
+#        \   ^__^
+#         \  (oo)\_______
+#            (__)\       )\/\
+#                ||----w |
+#                ||     ||
+#
+# Copyright 2024- the Outlines developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import math
+from collections import defaultdict
+from typing import DefaultDict, Dict, List, Optional
+
+import torch
+from pydantic import BaseModel
+
+try:
+    from outlines.fsm.fsm import RegexFSM
+    from outlines.fsm.json_schema import build_regex_from_schema
+except ImportError as e:
+    raise ValueError(
+        "Please install 'outlines' (pip install outlines) to use guided decoding."
+    ) from e
+
+
+class RegexLogitsProcessor:
```

Comment:
can you try adding a unit test for this now we are responsible for maintaining it?

Reply:
should it just test the class methods?

Reply:
Yeah like a new test that create the instance of RegexLogitsProcessor with tokenizer, and verify the input output and fsm transition happens correctly.